kfmon.c

```diff
@@ -1457,6 +1470,32 @@ static bool
 			if (thumbnails_count == 3U) {
 				is_processed = true;
 			}
+
+			if (thumbnails_count == 0U) {
+				char converted_book_path[CFG_SZ_MAX + 7];
```

Comment:
`KFMON_PATH_MAX` (or, I guess `sizeof(thumbnail_path)`, even)

Reply:
Which was a sensible decision ^^.

Or I guess another approach would just be to replace in-place, since it's only going to be 1:1 replacements?

Ooh, which raises a new question: does Nickel also does a byte-for-byte replacement, or does it do a per-codepoint replacement for multibyte glyphs?

Reply:
`file:///mnt/onboard/Books/The Will to Change_ Men, Masculinity, and Love -- bell hoo►ks -- 20$04 -- A§tria Booöküs -- 0743480333 -- fa╚8.epub`

becomes

`file____mnt_onboard_Books_The_Will_to_Change__Men,_Masculinity,_and_Love_--_bell_hoo►ks_--_20$04_--_A§tria_Booöküs_--_0743480333_--_fa╚8_epub`

The file system mounts with CP437 for me, afaik there is no multi byte characters in a fat filename, so that should not be an issue. I'll check your list of characters again, but it seems everything else is just fine.